Components/Queues/Drivers/QueueDriverInterface.php

```diff
@@ -41,72 +41,51 @@ public function disconnect();
 
     /**
-     * Returns true if a subscription already exists, false otherwise.
+     * Acknowledges the message in the message broker. $messageId is nullable for backwards compatibility with the
+     * SyncQueueDriverInterface. In practice, unless your driver is keeping track of messages, $message should always
+     * be passed to this function.
      *
-     * @return bool
+     * @param QueueMessageInterface|null $message
      */
-    public function isSubscribed();
+    public function ack(QueueMessageInterface $message = null);
 
     /**
-     * Creates a subscription to the given $queue, allowing to receive messages from it.
+     * Negative acknowledgement of a message.
      *
-     * @param string $queue Queue to subscribe
-     */
-    public function subscribe($queue);
-
-    /**
-     * Destroys the created subscription with a queue.
-     */
-    public function unSubscribe();
-
-    /**
-     * Acknowledges the processing of the last received object.
+     * @see ack() for extra information about this function.
      *
-     * The object should be removed from the queue.
+     * @param QueueMessageInterface|null $message
      */
-    public function ack();
-
-    /**
-     * Acknowledges a failure on processing the last received object.
-     *
-     * The object could be moved to the DLQ or be delivered to another subscription for retrial
-     */
-    public function nack();
+    public function nack(QueueMessageInterface $message = null);
 
     /**
      * @param QueueMessageInterface $message
-     * @param string|null           $destination
+     * @param string|null $destination
+     * @param array $arguments
      *
      * @return bool
      */
-    public function send(QueueMessageInterface $message, $destination = null);
-
-    /**
-     * Returns One Serializable object from the queue.
-     *
-     * It requires to subscribe previously to a specific queue
-     *
-     * @return \Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface|null
-     *
-     * @throws \Exception
-     */
-    public function receive();
+    public function send(QueueMessageInterface $message, $destination = null, array $arguments = []): bool;
 
     /**
      * @return QueueMessageInterface
      */
-    public function createQueueMessage();
+    public function createQueueMessage(): QueueMessageInterface;
 
     /**
-     * Clean all the opened resources, must be called just before terminating the current request.
+     * Returns the format used on serialize/deserialize function.
+     *
+     * @return string
      */
-    public function doDestroy();
+    public function getFormat(): string ;
 
     /**
-     * @return int The time it took in ms to de-queue and deserialize the message
+     * Set the format used on serialize/deserialize function.
+     *
+     * @param string|null $format
      */
-    public function getDequeueingTimeMs();
+    public function setFormat(string $format = null);
```

Comment:
In which case can we have a format null? If there is no case I would remove the null part from the interface

Reply:
In case when the user wants to declare a driver without the format. The default format is json, but we also have the array driver that should use other format.

Reply:
If a user wants to declare a driver without format, he wouldn't call to `setFormat`, it just I don't see people calling
```
$this->setFormat();
```
and not passing any variable, even if this variable the content is null

For this reason I was mentioning to remove the `= null` from the method definition

Reply:
Alright, I agree and did the change.

Reply:
I guess the change is not pushed yet, because I still see the `= null`. Also update all the classes that use this interface to remove the `= null` from the signature